From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 30 Jul 2016 09:55:52 +0200
Subject: [PATCH] {LLVMLinux} drivers/misc/lkdtm: handle clang
 -Wnull-dereference

clang warns when dereferencing the NULL pointers directly:

    drivers/misc/lkdtm_bugs.c:64:2: error: indirection of non-volatile
    null pointer will be deleted, not trap
          [-Werror,-Wnull-dereference]
            *((int *) 0) = 0;
            ^~~~~~~~~~~~

Use a volatile pointer instead.
---
 drivers/misc/lkdtm_bugs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/misc/lkdtm_bugs.c b/drivers/misc/lkdtm_bugs.c
index 182ae1894b32..804883c89474 100644
--- a/drivers/misc/lkdtm_bugs.c
+++ b/drivers/misc/lkdtm_bugs.c
@@ -61,7 +61,8 @@ void lkdtm_WARNING(void)
 
 void lkdtm_EXCEPTION(void)
 {
-	*((int *) 0) = 0;
+	volatile int *p = NULL;
+	*p = 0;
 }
 
 void lkdtm_LOOP(void)
-- 
